public/app/features/dashboard/state/PanelModel.ts

```diff
@@ -498,9 +502,26 @@ export class PanelModel implements DataConfigSource {
 
   replaceVariables(value: string, extraVars?: ScopedVars, format?: string) {
     let vars = this.scopedVars;
+
     if (extraVars) {
       vars = vars ? { ...vars, ...extraVars } : extraVars;
     }
+    const allVariablesParams = getAllVariableValuesForUrl(vars);
+    const variablesQuery = urlUtil.toUrlParams(allVariablesParams);
+    const timeRangeUrl = urlUtil.toUrlParams(getTimeSrv().timeRangeForUrl());
+
+    vars = {
+      ...vars,
+      [DataLinkBuiltInVars.keepTime]: {
+        text: timeRangeUrl,
+        value: timeRangeUrl,
+      },
+      [DataLinkBuiltInVars.includeVars]: {
+        text: variablesQuery,
+        value: variablesQuery,
+      },
+    };
```

Comment:
We should explore a way to do this lazily, but could be done in a separate PR to keep scope down for this